ql/src/java/org/apache/hadoop/hive/ql/exec/tez/TezSessionPool.java

```diff
@@ -395,20 +379,38 @@ int getInitialSize() {
     } while (!deltaRemaining.compareAndSet(oldVal, oldVal + delta));
     int toStart = oldVal + delta;
     if (toStart <= 0) return createDummyFuture();
-    LOG.info("Resizing the pool; adding " + toStart + " sessions");
-
-    // 2) If we need to create some extra sessions, we'd do it just like startup does.
-    int threadCount = Math.max(1, Math.min(toStart,
-        HiveConf.getIntVar(initConf, ConfVars.HIVE_SERVER2_TEZ_SESSION_MAX_INIT_THREADS)));
-    List<ListenableFutureTask<Boolean>> threadTasks = new ArrayList<>(threadCount);
-    // This is an async method, so always launch threads, even for a single task.
-    for (int i = 0; i < threadCount; ++i) {
-      ListenableFutureTask<Boolean> task = ListenableFutureTask.create(
-          new CreateSessionsRunnable(deltaRemaining));
-      new Thread(task, "Tez pool resize " + i).start();
-      threadTasks.add(task);
+    LOG.info("Resizing the pool; adding {} sessions", toStart);
+
+    final int threadCount =
+        Math.min(toStart, HiveConf.getIntVar(initConf, ConfVars.HIVE_SERVER2_TEZ_SESSION_MAX_INIT_THREADS));
+
+    return createSessions(toStart, threadCount);
+  }
+
+  private ListenableFuture<List<Boolean>> createSessions(int sessionCount, int maxParallel) {
+    Preconditions.checkArgument(sessionCount > 0);
+    Preconditions.checkArgument(maxParallel > 0);
```

Comment:
Second Precondition check is redundant

Reply:
This was previous checked in the calling methods.  I have moved it here to remove redundancy.  Not an expensive operation and it's not called all that often, so no worries even if it is doubled somewhere.  Better safe than sorry as the code changes.